### PR TITLE
Avoid segmentation fault.

### PR DESCRIPTION
**Pull request by ktlim**

### CODE REVIEW

File: src/mosaicfit.cc [hunk 1]
```diff
@@ -801,6 +801,9 @@ void KDTree::printSource() const {
 }
 
 KDTree::Ptr lsst::meas::mosaic::kdtreeMat(SourceMatchGroup &matchList) {
+    if (matchList.size() == 0) {
+        raise std::runtime_error("Can't create kd-tree for empty match list");
```
Comment (PaulPrice) [inline src/mosaicfit.cc]: Doesn't C++ use `throw`? (I'm surprised if this compiles.)
Any reason not to use an LSST exception like `lsst::pex::exceptions::RuntimeError`?

Comment (laurenam) [reply src/mosaicfit.cc]: I was gonna put it here 
https://github.com/lsst/meas_mosaic/commit/4bdc196d69919af5864bcb6e67ce9510cdce7a93

Comment (PaulPrice) [reply src/mosaicfit.cc]: You should do both.

Comment (laurenam) [reply src/mosaicfit.cc]: Looks like you’re suspicion was well-founded:
https://ci.lsst.codes/blue/organizations/jenkins/stack-os-matrix/detail/stack-os-matrix/28863/pipeline

Comment (ktlim) [reply src/mosaicfit.cc]: Sorry; I blew it.  Syntax highlighting should have told me.

Comment (ktlim) [reply src/mosaicfit.cc]: I agree that checking in both places is good.

**Comment by ktlim** [comment]
Now compiles, passes Jenkins, and:
```
>>> lsst.meas.mosaic.kdtreeMat([])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Can't create kd-tree for empty match list
```
as opposed to:
```
>>> lsst.meas.mosaic.kdtreeMat([])
Caught signal 11, backtrace follows:
```

(I could add a test, but I'd probably mess that up, too.)

**Pull request merged**